File: Classes/git/PBGitLane.m
//
//  PBGitLane.m
//  GitX
//
//  Created by Pieter de Bie on 27-08-08.
//  Updated to Objective-C implementation.
//

#import "PBGitLane.h"

@interface PBGitLane ()
@property (nonatomic, assign, readwrite) NSInteger index;
@end

@implementation PBGitLane

- (instancetype)initWithSHA:(NSString *)sha
{
    return [self initWithIndex:NSNotFound sha:sha];
}

- (instancetype)initWithIndex:(NSInteger)index sha:(NSString *)sha
{
    self = [super init];
    if (!self) {
        return nil;
    }

    _index = index;
    _sha = [sha copy];

    return self;
}

- (BOOL)isCommit:(NSString *)sha
{
    if (!sha.length || !self.sha.length) {
        return NO;
    }

    return [self.sha isEqualToString:sha];
}

@end

File: Classes/git/PBGitLane.swift
import Foundation

@objcMembers
@objc(PBGitLane)
final class PBGitLane: NSObject {
    @objc var sha: String?
    @objc let index: Int

    override init() {
        self.index = NSNotFound
        self.sha = nil
        super.init()
    }

    @objc(initWithSHA:)
    convenience init(sha: String?) {
        self.init(index: NSNotFound, sha: sha)
    }

    @objc(initWithIndex:sha:)
    init(index: Int, sha: String?) {
        self.index = index
        self.sha = sha
        super.init()
    }

    func isCommit(_ sha: String) -> Bool {
        guard !sha.isEmpty, let currentSHA = self.sha, !currentSHA.isEmpty else {
            return false
        }
        return currentSHA == sha
    }
}
